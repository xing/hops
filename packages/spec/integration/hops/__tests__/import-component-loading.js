import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import testRenderer from 'react-test-renderer';
import { App } from '../';

jest.mock('hops', () => {
  const hops = jest.requireActual('hops');

  hops.importComponent = () =>
    function ImportComponent({ render }) {
      return render({ loading: true });
    };

  return hops;
});

describe('ImportComponent loading', () => {
  it('should display the loading state', () => {
    const app = testRenderer.create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    expect(app).toMatchSnapshot();
  });
});
