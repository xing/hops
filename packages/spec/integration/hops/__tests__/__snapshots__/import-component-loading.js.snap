// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportComponent loading should display the loading state 1`] = `
<p>
  Fetching contentâ€¦
</p>
`;
